File: is_prime.cpp
#include <cmath>
#include <iostream>

bool isPrime(int n) {
  if (n < 2) {
    return false;
  }
  for (int i = 2; i <= std::sqrt(n); ++i) {
    if (n % i == 0) {
      return false;
    }
  }
  return true;
}

int main() {

    std::cout << isPrime(1) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(2) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(3) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(4) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(5) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(6) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(7) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(8) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(25) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(31) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(2017) << " == " << 1 << std::endl; // -> 1
    std::cout << isPrime(2048) << " == " << 0 << std::endl; // -> 0
    std::cout << isPrime(713) << " == " << 0 << std::endl; // -> 0

    return 0;
}

File: is_prime.py
from math import sqrt, floor

def is_prime(n):
  # not prime by default
  if n < 2:
    return False
  # sqrt is inflection point for factors
  for i in range(2, floor(sqrt(n)) + 1):
    if n % i == 0:
      return False
  return True

assert is_prime(1) == False
assert is_prime(2) == True 
assert is_prime(3) == True 
assert is_prime(4) == False
assert is_prime(5) == True 
assert is_prime(6) == False
assert is_prime(7) == True 
assert is_prime(8) == False
assert is_prime(25) == False
assert is_prime(31) == True
assert is_prime(2017) == True 
assert is_prime(2048) == False
assert is_prime(713) ==  False